lld/COFF/Writer.cpp

```diff
@@ -2688,6 +2693,21 @@ void Writer::prepareLoadConfig(SymbolTable &symtab, T *loadConfig) {
       loadConfig->DynamicValueRelocTableSection = relocSec->sectionIndex;
       loadConfig->DynamicValueRelocTableOffset =
           ctx.dynamicRelocs->getRVA() - relocSec->getRVA();
+
+      // On ARM64X, only the EC version of the load config contains
+      // CHPEMetadataPointer. Copy its value to the native load config.
+      if (!symtab.isEC() && ctx.hybridSymtab->loadConfigSize >=
```

Comment:
Can we assume that `ctx.hybridSymtab` exists here? What if dynamic relocs were to be used on a regular architecture?

Reply:
We currently create dynamic relocations only when the target is ARM64X, so checking `hybridSymtab` and `dynamicRelocs` amounts to the same thing. In theory, this could change in the future if we add support for other dynamic relocation types. I'll update it to use a separate branch to avoid relying on this assumption, which should also make the code a bit cleaner.

Reply:
Thanks! Yes that's what I meant - currently they're only used in ARM64X contexts, but in principle, dynamic relocations could maybe be used elsewhere as well.